File: 01_Backend/backend/src/main/java/org/example/backend/service/admin/AdminProductService.java
package org.example.backend.service.admin;

import org.example.backend.model.entity.admin.AdminProduct;
import org.example.backend.repository.admin.AdminProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class AdminProductService {

    @Autowired
    AdminProductRepository adminProductRepository;

    //    TODO: 전체조회(read)
    public List<AdminProduct> findAll() {
        List<AdminProduct> list = adminProductRepository.findAll();
        return list;
    }

    //    페이징 처리
    public Page<AdminProduct> findAllByAdminProductNameContaining(String pdName,
                                                                Pageable pageable) {
        Page<AdminProduct> page
                = adminProductRepository
                .findAllByAdminProductNameContaining(pdName, pageable);
        return page;
    }

    //    상세조회
    public Optional<AdminProduct> findById(Long pdId) {
        Optional<AdminProduct> optionalAdminProduct
                = adminProductRepository.findById(pdId);
        return optionalAdminProduct;
    }

    //    TODO: 등록(insert),수정(update)
    public AdminProduct save(AdminProduct adminProduct) {
        AdminProduct adminProduct2 = adminProductRepository.save(adminProduct);

        return adminProduct2;
    }

    //    TODO: 삭제(delete)
    public boolean removeById(Long pdId) {
        if (adminProductRepository.existsById(pdId) == true) {
            adminProductRepository.deleteById(pdId);
            return true;
        } else {
            return false;
        }
    }
}

File: frontend-vue/src/services/admin/AdminProductService.js
import http from "@/utils/http-common";
import LoginHeader from "../login/LoginHeader";

class AdminProductService {
    // 전체조회
    getAll(pdName, page, size) {
        return http.get(`/admin/admin-product?pdName=${pdName}&page=${page}&size=${size}`,{
            headers: LoginHeader()
        });
    }
    // 상세조회
    get(pdId) {
        return http.get(`/admin/admin-product/${pdId}`,{
            headers: LoginHeader()
        });
    }
    // 상품등록함수
    create(data) {
        return http.post("/admin/admin-product", data,{
            headers: LoginHeader()
        });
    }
    // 상품수정함수
    update(pdId, data) {
        return http.put(`/admin/admin-product/${pdId}`, data,{
            headers: LoginHeader()
        });
    }
    // 상품삭제함수
    delete(pdId) {
        return http.delete(`/admin/admin-product/deletion/${pdId}`,{
            headers: LoginHeader()
        });
    }
}

export default new AdminProductService();